shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -370,10 +370,13 @@
   for (int64_t view_id : composition_order_) {
     EnsureOverlayInitialized(view_id, gl_context, gr_context);
     auto frame = overlays_[view_id]->surface->AcquireFrame(frame_size_);
-    SkCanvas* canvas = frame->SkiaCanvas();
-    canvas->drawPicture(picture_recorders_[view_id]->finishRecordingAsPicture());
-    canvas->flush();
-    did_submit &= frame->Submit();
+    if (frame) {
+      // We should already be printing an error if we got here.
```

Comment:
I'm a little confused by this comment, can you elaborate a bit more?

Reply:
`AcquireFrame` returning null is an error.  From my experience, if it returns null, it prints out an error message.  I don't want to print out 2 error messages for the same error.

Reply:
For example: https://github.com/flutter/engine/blob/a86ef946563b020108320bbfb974bf7343284fd3/shell/gpu/gpu_surface_gl.cc#L264

Reply:
I updated it, it was in the wrong place.

Reply:
Thanks makes more sense to me now.